### PR TITLE
Cv 67 use url helper

### PR DESCRIPTION
**Pull request by iangledhill**
Extend the Url Helper to provide "reservations links" and include this in the CV-08 links (manager reservations).
Refactor URL helper to be an injectable service (so that can be used in controllers). 
Required package upgrades to get the 2.0.1 of the Auto Configuration package.

### CODE REVIEW

File: src/SFA.DAS.ProviderUrlHelper/LinkGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        public string ReservationsLink(string path)
+        {
+            var configuration = _lazyProviderConfiguration.Value;
+            var baseUrl = configuration.ReservationsBaseUrl;
+
+            return Action(baseUrl, path);
+        }
+
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderUrlHelper/LinkGenerator.cs]: The 3 actions above are identical apart from the Configuration Property. Would it be worthwhile writing a private method which accepts the property and path variables and returns the action link?

File: src/SFA.DAS.ProviderUrlHelper/Framework/ServiceLocator.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿#if NETFRAMEWORK
+using System.Web.Mvc;
+
+namespace SFA.DAS.ProviderUrlHelper.Framework
+{
+    public static class ServiceLocator
+    {
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderUrlHelper/Framework/ServiceLocator.cs]: Just an thought, is it worthwhile having this class as internal?

File: src/SFA.DAS.ProviderUrlHelper/Core/ServiceLocator.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿#if NETCOREAPP
+using Microsoft.AspNetCore.Http;
+
+namespace SFA.DAS.ProviderUrlHelper.Core
+{
+    public static class ServiceLocator
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderUrlHelper/Core/ServiceLocator.cs]: Just an thought, is it worthwhile having this class as internal?

**Comment by iangledhill** [comment]
Abandoned

**Pull request closed without merging**